server/routes/changeGoal/ChangeGoalController.ts

```diff
@@ -34,9 +37,24 @@ export default class ChangeGoalController {
       const selectedAreaOfNeed = sortedAreasOfNeed.find(areaOfNeed => areaOfNeed.name === goal.areaOfNeed.name)
       const form = errors ? req.body : this.mapGoalToForm(goal)
 
+      const criminogenicNeedsData = req.services.sessionService.getCriminogenicNeeds()
       const planUuid = req.services.sessionService.getPlanUUID()
       const plan = await req.services.planService.getPlanByUuid(planUuid)
 
+      let assessmentDetailsForArea = null
+
+      // get assessment data
+      try {
+        const assessmentResponse = await req.services.assessmentService.getAssessmentByUuid(planUuid)
+        const assessmentData = assessmentResponse.sanAssessmentData
+
+        // 2. get the assessment details for the selected area of need
+        const areaConfig: AssessmentAreaConfig = areaConfigs.find(config => config.area === goal.areaOfNeed.name)
+        assessmentDetailsForArea = getAssessmentDetailsForArea(criminogenicNeedsData, areaConfig, assessmentData)
+      } catch {
+        /* swallow the error, missing data is handled in the template */
+      }
```

Comment:
Could we avoid using 'let' by writing this block in a more functional way? Something like:

const assessmentDetailsForArea = await req.services.assessmentService
.getAssessmentByUuid(planUuid)
.then(assessmentResponse => {
 const assessmentData = assessmentResponse.sanAssessmentData
 const areaConfig: AssessmentAreaConfig = areaConfigs.find(config => config.area === goal.areaOfNeed.name)
 return getAssessmentDetailsForArea(criminogenicNeedsData, areaConfig, assessmentData)
})
 .catch(() => null)

Reply:
@JakeGoodwin22 resolved in https://github.com/ministryofjustice/hmpps-sentence-plan-ui/pull/393/commits/7f467f893e3f9244d50ac8be87f2fcc31f41b5ab